src/Types/HashBytes.cs

```diff
@@ -0,0 +1,13 @@
+using Google.Protobuf;
+
+namespace Concordium.Sdk.Types;
+
+/// <summary>
+/// A general wrapper of a 32-size hash. This is used to add type safety to
+/// a hash that is used in different context.
+/// </summary>
+public record HashBytes : Hash
```

Comment:
I'm not sure I understand how this adds type safety. In the rust world we use a phantom type parameter for this, but here there are no type parameters. So how does this add safety?

Reply:
Made types explicit